### PR TITLE
[Fixes AB#559] Wallet redesign

### PR DESCRIPTION
**Pull request by Skafec**
Wallet screen complete overhaul.

Added Font Awesome Pro to the project (do npm install after branch pull and check
if all icons are shown properly)

Created new partial _pagination to control wallet transaction history pagination (this
is normally controlled via Bootstrap)

### CODE REVIEW

File: src/app/wallet/wallet.component.html [hunk 1]
```diff
@@ -43,94 +43,160 @@ <h4 class="mb-3 text-uppercase">
...
+                    <div *ngIf="transactionHistory.length === 0">
+                        <span class="transactions__empty"></span>Your transaction history is empty.
+                    </div>
+                    <div *ngIf="transactionHistory !== undefined && transactionHistory.length > 0" 
+                         class="wow fadeIn">
+                        <div class="transactions__history">
+                            <ng-container *ngFor="let transaction of transactionHistoryPage">
+                                <ng-container *ngIf="shouldShowTransaction(transaction)">
+                                    <div class="transactions__single-element" 
+                                         [class.transactions__error] = "transaction.state !== 'MINED' || transaction.type === 'APPROVE_INVESTMENT'">
```
Comment (mightymatth) [inline src/app/wallet/wallet.component.html:132]: why this logic? shouldn't it be `transaction.state === 'FAILED'`?

File: src/app/wallet/wallet.component.html [hunk 2]
```diff
@@ -43,94 +43,160 @@ <h4 class="mb-3 text-uppercase">
...
+                                                    Dividend <span class="left__conjuctor">from</span>
+                                                </div>
+                                                <div class="left__transaction-type" 
+                                                    *ngIf="transaction.type === 'CANCEL_INVESTMENT'">
+                                                    Cancel investment <span class="left__conjuctor">to</span>
+                                                </div>
+                                                <div class="left__transaction-type" 
+                                                    *ngIf="transaction.type === 'APPROVE_INVESTMENT'">
+                                                    Approve Investment <span class="left__conjuctor">to</span>
+                                                </div>
```
Comment (mightymatth) [inline src/app/wallet/wallet.component.html:158]: too much code for a simple mapping that can be done in a pipe.

File: src/app/wallet/wallet.component.html [hunk 3]
```diff
@@ -43,94 +43,160 @@ <h4 class="mb-3 text-uppercase">
...
+                                                {{ transaction.description }}
+                                            </div>
+                                            <div class="left__info">
+                                                <span class="info__state-mined" 
+                                                    *ngIf="transaction.state === 'MINED'">
+                                                    <span class="state__invested" 
+                                                            *ngIf="transaction.type === 'INVEST'">
+                                                        1% Invested -
```
Comment (mightymatth) [inline src/app/wallet/wallet.component.html:166]: When something is hardcoded, like this 1%, a good practice is to set a comment starting with TODO, e.g.:

```
// TODO: change this when percentage will be available from API
```

File: src/app/wallet/wallet.component.html [hunk 4]
```diff
@@ -43,94 +43,160 @@ <h4 class="mb-3 text-uppercase">
...
+                                                </div>
+                                                <div class="left__transaction-type" 
+                                                    *ngIf="transaction.type === 'APPROVE_INVESTMENT'">
+                                                    Approve Investment <span class="left__conjuctor">to</span>
+                                                </div>
+                                                {{ transaction.description }}
+                                            </div>
+                                            <div class="left__info">
+                                                <span class="info__state-mined" 
+                                                    *ngIf="transaction.state === 'MINED'">
```
Comment (mightymatth) [inline src/app/wallet/wallet.component.html:163]: This is not meant to be used like that.
transaction.state is an enum, so it should be compared with enums. You can access TransactionState in template by making a varible inside component:

```
transactionState = TransactionState
```

then in template, you use it like:

```
<div *ngIf="transaction.state === transactionState.MINED">...</div>
```

This approach is more secure and less prone to human errors.

**Review by mightymatth — CHANGES_REQUESTED** [review]
I'll fix the issues mentioned below and implement the investment percentage (it's been recently developed on the backend side).
Just read the comments for info.

File: src/app/wallet/wallet.component.html (hunk 1 above)
Comment (mightymatth) [reply src/app/wallet/wallet.component.html:132]: It's OK. nvm! :)

File: src/app/wallet/wallet.component.html (hunk 2 above)
Comment (Skafec) [reply src/app/wallet/wallet.component.html:158]: I agree, wanted to ask you if there was any other way to write this logic without all of the unnecessary conditions but I forgot 😄

File: src/app/wallet/wallet.component.html (hunk 4 above)
Comment (Skafec) [reply src/app/wallet/wallet.component.html:163]: Oh right, thanks for that info! 😄

File: src/app/wallet/wallet.component.scss [hunk 5]
```diff
@@ -111,20 +111,14 @@
...
     &__desc {
       line-height: 20px;
       letter-spacing: 0.18px;
     }
-    &__transaction-type {
-      font-weight: bold;
-    }
-    &__conjuctor {
-      display: none;
-    }
```
Comment (Skafec) [inline src/app/wallet/wallet.component.scss]: This was used to hide conjuctors on mobile.
If you look at design, these are hidden on mobile version

File: src/app/wallet/wallet.component.html [hunk 6]
```diff
@@ -129,54 +129,44 @@ <h4 class="init-left__title">
...
                                                     Pending
                                                 </span>
-                                                <span class="info__state-failed" 
-                                                    *ngIf="transaction.state === 'FAILED'">
+                                                <span class="info__state-failed" *ngIf="transaction.state === 'FAILED'">
                                                     Failed
-                                                    <span class="state__invested" 
-                                                            *ngIf="transaction.type === 'INVEST'">
-                                                        1% Invested -
-                                                    </span>
```
Comment (Skafec) [inline src/app/wallet/wallet.component.html]: We have % date on failed investment in design, shold this be removed?

Also, why didn't you use transactionState.PENDING and transactionState.FAILED above?

**Pull request merged**